third_party/terraform/tests/resource_dataflow_job_test.go

```diff
@@ -182,6 +184,35 @@ func TestAccDataflowJob_withIpConfig(t *testing.T) {
 	})
 }
 
+func TestAccDataflowJob_streamUpdate(t *testing.T) {
+	t.Parallel()
+
+	randStr := acctest.RandString(10)
+	bucket := "tf-test-dataflow-gcs-" + randStr
+	job := "tf-test-dataflow-job-" + randStr
+	topic := "tf-test-dataflow-topic-" + randStr
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDataflowJobDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataflowJob_updateStream(bucket, job, topic, "folder1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDataflowJobExists(t, "google_dataflow_job.pubsub_stream"),
+				),
+			},
+			{
```

Comment:
Do you know whether the `job_id` is retained after updating? If it is, you could do something like  https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/tests/resource_monitoring_slo_test.go#L94 to ensure that the update worked (and a similar thing for a BATCH job in a different test to ensure it `ForceNew`s)

Reply:
No, the replacement job will have a new `job_id`.

Reply:
Modified the test to check for updates in the `temp_gcs_location` field